DatabaseDefinition.Test/DatabaseDefinitionTestBase.cs

```diff
@@ -0,0 +1,98 @@
+﻿using DatabaseDefinition.EntityModel;
+using DatabaseDefinition.EntityModel.Repositories;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.EntityFrameworkCore.Storage;
+using TM = TransferModel;
+
+namespace DatabaseDefintion.Test
+{
+    /// <summary>
+    /// This class provides repositories with an in-memory db.
+    /// The in-memory db saves an update partially, if the commit failed.
+    /// The in-memory db also does not set db-generated values.
+    /// The in-memory db has other behaviors differing from a real db.
+    /// For these reasons, test cases may require a real local db for testing.
+    /// The huge benefit of the in-memory db is that is provided for testing very quick.
+    /// The local db enables true db logic testing, but comes with a high performance cost.
+    /// If we want to use those in the future we need to add a way to choose whether a test should use a in-memory db or a test sql db.
+    /// </summary>
+    public class DatabaseDefinitionTestBase
+    {
+        protected ItemRepository ItemRepository { get; }
+        protected ItemPictureRepository ItemPictureRepository { get; }
+        protected CancellationToken CancellationToken { get; } = CancellationToken.None;
```

Comment:
Sollte man den hier anlegen oder in jede Async Methode übergeben?

Reply:
Es würde glaube ich keinen Vorteil bringen, die in jede Async Methode extra reinzugeben, da die Methoden nur für die Unit-Tests benutzt werden.